app/utils/metadata.py

```diff
@@ -569,506 +569,309 @@ def __update_from_settings(self) -> None:
             log_deps_order_info(self.internal_local_metadata)
         else:
             logger.info("No Steam database supplied from external metadata. skipping.")
-        # Add load order to installed mods based on dependencies from community rules
-        if self.external_community_rules:
-            logger.info("Started compiling metadata from configured Community Rules")
-            for package_id in self.external_community_rules:
-                # Note: requiring the package be in self.internal_local_metadata should be fine, as
-                # if the mod doesn't exist self.internal_local_metadata, then either mod_data or dependency_id
-                # will be None, and then we don't insert a dependency
+
+        def process_external_rules(rules, rule_type, opposite_rule_type):
+            for package_id, rule_data in rules.items():
                 if package_id.lower() in self.packageid_to_uuids:
                     potential_uuids = self.packageid_to_uuids.get(
                         package_id.lower(), set()
                     )
-                    load_these_after = self.external_community_rules[package_id].get(
-                        "loadBefore"
-                    )
-                    if load_these_after:
-                        logger.debug(
-                            f"Current mod should load before these mods: {load_these_after}"
-                        )
-                        # In Alphabetical, load_these_after is at least an empty dict
-                        # Cannot call add_load_rule_to_mod outside of this for loop,
-                        # as that expects a list
-                        for load_this_after in load_these_after:
-                            for uuid in potential_uuids:
-                                add_load_rule_to_mod(
-                                    self.internal_local_metadata[
-                                        uuid
-                                    ],  # Already checked above
-                                    load_this_after,  # Lower() done in call
-                                    "loadTheseAfter",
-                                    "loadTheseBefore",
-                                    self.internal_local_metadata,
-                                    self.packageid_to_uuids,
-                                )
-                    load_these_before = self.external_community_rules[package_id].get(
-                        "loadAfter"
-                    )
-                    if load_these_before:
-                        logger.debug(
-                            f"Current mod should load after these mods: {load_these_before}"
-                        )
-                        # In Alphabetical, load_these_before is at least an empty dict
-                        for load_this_before in load_these_before:
+                    load_these = rule_data.get(rule_type)
+                    if load_these and isinstance(load_these, list):
```

Comment:
Just use static type checking if the param is supposed to only be one type

Reply:
Fixed